packages/instructions/src/roles.ts

```diff
@@ -0,0 +1,59 @@
+export enum AccountRole {
+    WRITABLE_SIGNER = 0b1, // 1
+    READONLY_SIGNER = 0b10, // 2
+    WRITABLE = 0b100, // 4
+    READONLY = 0b1000, // 8
+}
```

Comment:
I'm probably missing something, but is there a reason for defining each role as a separate bit rather than as just a signer bit and a writable bit? ie:

```
export enum AccountRole {
    READONLY = 0b00, // 0
    WRITABLE = 0b01, // 1
    READONLY_SIGNER = 0b10, // 2
    WRITABLE_SIGNER = 0b11, // 3
}
```

Then you don't need to do any of the `<` and `>` checks or bit shifts and just use `&` and `|` everywhere

Reply:
That seems less dumb. Let me refactor that and if I don't discover a reason why I did it this way and not the way you've described, I'll ship it.

Hooray for test coverage that lets me do this with reckless abandon.